src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+            this.storage = new Storage();
+            this.tasks = new TaskList(storage.readTasks());
+            this.parser = new Parser();
+
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        }
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        try {
+            CommandType commandType = parser.parseCommandType(input);
+            return handleCommand(commandType, input);
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        }
+        return "Duke heard: " + input;
+    }
+
+    private void run() {
+        ui.printWelcomeMessage();
+        try (Scanner sc = new Scanner(System.in)) {
+            while (true) {
+                String input = sc.nextLine();
+                if (input.equals("bye")) {
+                    break;
+                }
+                CommandType commandType = parser.parseCommandType(input);
+                handleCommand(commandType, input);
+            }
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        } catch (Exception e) {
+            ui.printErrorMessage(new DukeException("An unexpected error occurred: " + e.getMessage()));
+        }
+        ui.printFarewellMessage();
+    }
+
+    /**
+     * Main entry point of the Duke application.
+     * @param args Command line arguments.
+     */
+    public static void main(String[] args) {
+        Duke ekud = new Duke();
+        ekud.run();
+    }
+
+    // private void handleCommand(CommandType commandType, String command) throws DukeException {
+    //     switch (commandType) {
+    //     case LIST:
+    //         ui.printList(tasks.getTasks());
+    //         break;
+    //     case MARK:
+    //         markTask(command);
+    //         break;
+    //     case DELETE:
+    //         deleteTask(command);
+    //         break;
+    //     case TODO:
+    //     case DEADLINE:
+    //     case EVENT:
+    //         addTask(command);
+    //         break;
+    //     case FIND:
+    //         handleFind(command);
+    //         break;
+    //     case UNKNOWN:
+    //         ui.printErrorMessage(new DukeException("I'm sorry, but I don't know what that means :-("));
+    //         break;
+    //     }
+    // }
+
+    private String handleCommand(CommandType commandType, String command) throws DukeException {
+        Task task;
+        switch (commandType) {
+        case LIST:
+            ui.printList(tasks.getTasks());
+            return ui.displayList(tasks.getTasks());
+
+        case MARK:
+            task = markTask(command);
+            return ui.displayMarkedTaskConfirmation(task);
+
+        case DELETE:
+            task = deleteTask(command);
+            return ui.displayDeletedTaskConfirmation(task, tasks);
+
+        case TODO:
+        case DEADLINE:
```

Comment:
Should comment // Fallthrough for TODO and DEADLINE cases.

Reply:
This was an intentional choice, as EVENT, TODO and DEADLINE runs through the same event handler addTask(command)